RecoDataProducts/inc/LumiInfo.hh

```diff
@@ -0,0 +1,49 @@
+//
+// Class to collect the info needed for monitoring POT / stop muons
+// bvitali May 2021
+//
+
+
+#ifndef RecoDataProducts_LumiInfo_hh
+#define RecoDataProducts_LumiInfo_hh
+
+namespace mu2e {
+
+    class LumiInfo
+    {
+        public:
+            LumiInfo(): nTrackerHits_(0), nCaloHits_(0),
+                        nProtonTCs_(0), caloEnergy_(0),
+                        testVariable_(0)
+            {}
+
+            LumiInfo(unsigned short nTrackerHits, unsigned short nCaloHits,
+                    unsigned short nProtonTCs, unsigned short caloEnergy,
+                    unsigned short testVariable):
+                nTrackerHits_(nTrackerHits), nCaloHits_(nCaloHits),
+                nProtonTCs_(nProtonTCs), caloEnergy_(caloEnergy),
+                testVariable_(testVariable)
+            {}
+
+            void setnTrackerHits   (unsigned short tmp) {nTrackerHits_ = tmp;}
+            void setnCaloHits      (unsigned short tmp) {nCaloHits_    = tmp;}
+            void setnProtonTCs     (unsigned short tmp) {nProtonTCs_   = tmp;}
+            void setcaloEnergy     (unsigned short tmp) {caloEnergy_   = tmp;}
+            void settestVariable   (unsigned short tmp) {testVariable_ = tmp;}
```

Comment:
My comment got lost, trying again.     I am asking to delete these 5 lines - the modifier methods.  Having setters is a common anti-pattern.  Instead collect all the info you need and then instantiate the object so that everthing is initialized in the constructor.  That makes the code cleaner and easier to follow.   Once there is a setter one can never be sure where a value of an object field came from...

Reply:
In most cases Andrei is absolutely right.  There are some cases, usually big complicated objects that require a multi-stage build, for which there can be exceptions.   It seems unlikely that this is a good case for an exception

Reply:
Yes, I second that. This is also less error prone because you can't forget to set something at some point.

Reply:
I do see your point but isn't useful to be able to change a value without having to set all the others?
Let's say I'm using just the nProtonTCs: I need to remember which entries is "nTrackersHits" and set the others to 0.
Instead now I could just saying:
LumiInfo lum; lum.setnProtonTCs(4);

Or am I mistaken? Either case sure I can remove those lines. I just thought would be handy.